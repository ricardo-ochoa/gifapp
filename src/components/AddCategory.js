import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const  AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );
    }

    const handleSumit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2){
            setCategories( cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit = { handleSumit }>
            <h2>Add Category</h2>
            <input
                type='text'
                value={ inputValue }
                onChange={ handleInputChange }
            ></input>
        </form>
    )
}

AddCategory.prototypes = {
    setCategories: PropTypes.func.isRequired
}