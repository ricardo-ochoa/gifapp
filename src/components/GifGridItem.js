import React from 'react'
import PropTypes from 'prop-types';


export const GifGridItem = ({id, url, title }) => {


    return (
        <>
            <div className="card animate__animated animate__pulse">
                <img src= {url} alt={ title }></img>
                      
            </div>
            
        </>
    )
}

GifGridItem.prototypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}
