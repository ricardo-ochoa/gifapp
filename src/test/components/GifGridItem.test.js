import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

const title = "One Piece";
const url = "https://giphy.com/onepiece.jpg"

const wrapper = shallow(<GifGridItem title = { title } url = { url }/>);

describe('Pruebas en GifGridItem', () =>{

    test('Debe mostrarse correctamente <GifGridItem />', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe mostrarse correctamente la imagen y alt', () => {

        const img = wrapper.find('img');
        expect( img.prop( 'src' ) ).toBe( url );
        expect( img.prop( 'alt' ) ).toBe( title );
    });

    test('Debe de tener animate_pulse', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes( 'animate__pulse' ) ).toBe( true );
    });
    
})