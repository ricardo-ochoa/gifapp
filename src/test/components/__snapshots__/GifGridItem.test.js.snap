// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrarse correctamente <GifGridItem /> 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__pulse"
  >
    <img
      alt="One Piece"
      src="https://giphy.com/onepiece.jpg"
    />
  </div>
</Fragment>
`;
